components/portal_players/commons/portal_players.lua

```diff
@@ -0,0 +1,279 @@
+		conditionString = conditionString .. ' AND ' .. gameConditions
+	elseif String.isNotEmpty(gameConditions) then
+		conditionString = gameConditions
+	end
+
+	local players = mw.ext.LiquipediaDB.lpdb('player', {
+		query = 'pagename, id, name, team, status, type, extradata, links, nationality, localizedname, birthdate, deathdate',
+		order = 'id asc',
+		conditions = conditionString,
+		limit = 5000,
+	})
+
+	return PortalPlayers._groupPlayerData(players)
+end
+
+---Retrieves the country list.
+---@param regionsInput string?
+---@param countriesInput string?
+---@param gameConditions string?
+---@return string[], string[]
+function PortalPlayers._getCountries(regionsInput, countriesInput, gameConditions)
+	local regionConditions = String.isNotEmpty(regionsInput) and
+		Array.map(Array.map(mw.text.split(regionsInput --[[@as string]], ',', true), String.trim), function (region)
+			return '[[region::' .. region .. ']]'
+		end) or {}
+
+	if String.isNotEmpty(countriesInput) then
+		---@cast countriesInput -nil
+		local countries = Array.map(mw.text.split(countriesInput, ',', true), String.trim)
+		if Table.isNotEmpty(countries) then
+			return countries, regionConditions
+		end
+	end
+
+	local conditionString = Table.isNotEmpty(regionConditions)
+		and ('(' .. table.concat(regionConditions, ' OR ') .. ')') or ''
+
+	if String.isNotEmpty(conditionString) and String.isNotEmpty(gameConditions) then
+		conditionString = conditionString .. ' AND ' .. gameConditions
+	elseif String.isNotEmpty(gameConditions) then
+		---@cast gameConditions -nil
+		conditionString = gameConditions
+	end
+
+	local queryData = mw.ext.LiquipediaDB.lpdb('player', {
+		query = 'nationality',
+		groupby = 'nationality asc',
+		order = 'nationality asc',
+		conditions = conditionString,
+		limit = 5000,
+	})
+
+	return Array.map(queryData, function(item) return item.nationality end), regionConditions
+end
+
+---Groups the "player" data by country and wether they are players or not
+---@param players table[]
+---@return {[string]: {players: table[], nonPlayers: table[]}}
+function PortalPlayers._groupPlayerData(players)
+	local _, groupedByCountry = Array.groupBy(players, function(player) return player.nationality --[[@as string]] end)
+
+	return Table.map(groupedByCountry, function(country, countryPlayerData)
+		local groupedData
+		_, groupedData = Array.groupBy(countryPlayerData, function(player)
+			local extradata = player.extradata or {}
+			return Logic.nilOr(
+				Logic.readBoolOrNil(extradata.isplayer),
+				(player.type or ''):lower() == 'player' or (extradata.role or ''):lower() == 'player'
+			) and 'players' or 'nonPlayers'
+		end)
+		---@cast groupedData {players: table[], nonPlayers: table[]}
+		return country, groupedData
+	end)
+end
+
+---Builds the table display for a given set of players
+---@param playerData table[]?
+---@param flag string
+---@param playerType string?
+---@return Html?
+function PortalPlayers:buildCountryTable(playerData, flag, playerType)
```

Comment:
boolean or enum makes more sense for playerType than a string

Reply:
the idea was to be able to overwrite the `'Player'` text for e.g. f1

Reply:
Don't see the reason as long it's two categories ("driver" and "non-driver")? Still a boolean isPlayer works just. If there's more than 2 options needed, then make it an enum

Reply:
will probably rework it to not use the same var for 2 purposes